packages/sdk/src/browser-wallets/okx/index.ts

```diff
@@ -0,0 +1,152 @@
+import { Psbt } from "bitcoinjs-lib";
+
+import { getAddressFormat } from "../../addresses";
+import { BrowserWalletNetwork } from "../../config/types";
+import {
+  BrowserWalletExtractTxFromNonFinalizedPsbtError,
+  BrowserWalletNotInstalledError,
+  OrditSDKError,
+} from "../../errors";
+import { BrowserWalletSignResponse, WalletAddress } from "../types";
+import { OKXSignPSBTOptions } from "./types";
+
+function isInstalled(): boolean {
+  if (typeof window === "undefined") {
+    throw new OrditSDKError("Cannot call this function outside a browser");
+  }
+  return typeof window.okxwallet !== "undefined";
+}
+
+async function getOKXWalletProvider(
+  network: BrowserWalletNetwork = "mainnet",
+): Promise<OKXWalletProvider> {
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("OKX Wallet not installed.");
+  }
+
+  const provider =
+    network === "mainnet"
+      ? window.okxwallet.bitcoin
+      : window.okxwallet.bitcoinTestnet;
+
+  if (!provider) {
+    throw new OrditSDKError("Failed to get OKX Wallet provider.");
+  }
+
+  return provider;
+}
+
+async function getAddresses(
+  network: BrowserWalletNetwork = "mainnet",
+): Promise<WalletAddress[]> {
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("OKX Wallet not installed.");
+  }
+
+  const provider = await getOKXWalletProvider(network);
+
+  const { address, publicKey } = await provider.connect();
```

Comment:
I think we will need to add try-catch on this line to ensure that we throw request cancelled by user error

Reply:
Added error handling for all methods